transport/server_test.go

```
@@ -185,3 +192,99 @@ func TestServer(t *testing.T) {
+	newSessionCalled := false
+	handler := newWSHandler(server, func(Session) error {
+		newSessionCalled = true
+		return errors.New("self-connect request should not create session")
+	})
+
+	conn := &selfConnectConn{addr: &net.TCPAddr{IP: net.ParseIP("127.0.0.1"), Port: 65000}}
+	rw := &hijackResponseWriter{
+		header: make(http.Header),
+		conn:   conn,
+	}
+	req, err := http.NewRequest(http.MethodGet, "http://127.0.0.1/ws", nil)
```

golangci-lint (2.12.2)

[error] net/http.NewRequest must not be called. use net/http.NewRequestWithContext

(noctx)